### DISCUSSION TITLE
다음 테스트 코드는 어떤 문제가 있을까요?

### DISCUSSION THREAD

**Discussion by le2sky** [2024-woowacourse-study/level-interview#48] [Test]
```java

@Test
void add() {
    Calculator calculator = new Caulculator();
    int value1 = 1;
    int value2 = 2;

    int actual = calculator.add(value1,  value2);

    int expected = value1 + value2;
    assertThat(actual).isEqualTo(expected);
}

```

**Comment by gyungchan-jo** [comment]
expected 변수가 `Calculator` 클래스 `add()` 메서드의 내부 구현을 알고 테스트하는 것이 문제입니다.

이 문제를 해결하기 위해 expected 변수를 지우고 `assertThat(actual).isEqualTo(3);` 다음과 같이 바꾸는 것이 좋습니다.

**Comment by le2sky** [comment]
`int expected = value1 + value2;`

부분이 문제라고 생각합니다. 테스트로 도메인 지식이 유출되었기 때문입니다.
Calculator.add 메서드의 로직이 테스트 코드로 유출되었습니다.
이러한 경우, 구현 세부 사항과 결합된 것과 마찬가지입니다.
올바르게 테스트하기 위해서는 expected 값을 하드 코딩하는게 차라리 좋습니다.
